### PR TITLE
Put internets using validation behind an interface

### PR DESCRIPTION
**Pull request by JeroenDeDauw**
Still working on removing the internets usage from the tests

### CODE REVIEW

**Comment by JeroenDeDauw** [comment]
Featuring: anonymous class

anonymous

**Comment by gbirke** [comment]
+1

File: src/Validation/MailValidator.php [hunk 1]
```diff
@@ -13,14 +14,11 @@
...
-	private $testWithMX;
+	private $domainValidator;
 	private $lastViolation;
 
-	const TEST_WITH_MX = true;
-	const TEST_WITHOUT_MX = false;
-
-	public function __construct( bool $testWithMX ) {
-		$this->testWithMX = $testWithMX;
+	public function __construct( DomainNameValidator $tldValidator ) {
```
Comment (KaiNissen) [inline src/Validation/MailValidator.php]: Not really an issue, but it's actually not a top level validator.

**Comment by KaiNissen** [comment]
Apart from my picky comment, also +1

File: src/Validation/MailValidator.php (hunk 1 above)
Comment (JeroenDeDauw) [inline src/Validation/MailValidator.php]: Yeah, I forgot to rename this var. Originally I thought this stuff only did TLD validation.

**Pull request merged**